vignettes/target_types.Rmd

```diff
@@ -0,0 +1,49 @@
+---
+title: "Target type guidelines"
+author: "David Watkins"
+date: "3/25/2019"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Target types}
+  %\VignetteEngine{knitr::rmarkdown}
+  \usepackage[utf8]{inputenc}
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+`scipiper` targets can be indicator files with a [shared cache](shared_cache.html), local data files with or without local indicator files, or simply R objects.  This vignette provides some guidelines for when to use these different patterns.
+
+## Objectives
+There are two main goals to keep in mind when deciding what style of target to use:
+ 
+ - **Minimize time spend unnecessarily rebuilding targets** --- i.e., those that have already been built locally or by collaborators
+ - **Keeping the pipeline simple** --- caching can save lots of build time, but also adds to the code that has to be maintained, can clutter the repo with build and indicator files, etc.
+ 
+ These two can sometimes be in tension, but usually it will be clear which is more important for a given case.
```

Comment:
Nice distillation.

Reply:
...and now that I've read and thought through the full vignette, I'm wondering, which of these considerations applies when choosing between a local data file and a local R object? Is there a third consideration that needs to be added?

Reply:
Added **Do you need easy access to the underlying file**